BoxBrowseSDK/BoxBrowseSDK/Controllers/BOXItemsViewController.m

```diff
@@ -50,17 +50,23 @@ - (void)viewWillAppear:(BOOL)animated
     [super viewWillAppear:animated];
     
     if (self.contentClient.user == nil) {
-        BOXAuthorizationViewController *authViewController = [[BOXAuthorizationViewController alloc] initWithSDKClient:self.contentClient completionBlock:^(BOXAuthorizationViewController *authorizationViewController, BOXUser *user, NSError *error) {
-            [authorizationViewController dismissViewControllerAnimated:YES completion:^{
-                [self refresh];
+        // if for some reason the authentication fails, the dismissal of the BOXAuthorizationViewController will cause BOXItemsViewController's viewWillAppear to get called.
+        // Since self.contentClient.user would still be nil (auth failed), this would enter a loop to present the auth controller again.
+        if (self.authenticationAttempted == NO) {
+            BOXAuthorizationViewController *authViewController = [[BOXAuthorizationViewController alloc] initWithSDKClient:self.contentClient completionBlock:^(BOXAuthorizationViewController *authorizationViewController, BOXUser *user, NSError *error) {
```

Comment:
Couple notes from chatting offline:
1. We should look into why the cancel button is not being shown in the auth VC
2. In app-users mode, the developer would not want an Auth VC to be presented. In that scenario, we should probably just attempt a currentUserRequest and then show an error if there is a failure.